torch/amp/grad_scaler.py

```diff
@@ -0,0 +1,700 @@
+from __future__ import annotations
```

Comment:
Can you publish a diff showing what the additional changes on this file are?

Reply:
Here is the diff file [grad_scaler.txt](https://github.com/pytorch/pytorch/files/13988657/grad_scaler.txt)